### DISCUSSION TITLE
Is vscode.window.createTreeView async?

### DISCUSSION THREAD

**Discussion by McSodbrenner** [microsoft/vscode-discussions#125] [Extension Development QnA] [answered]
Hi, I could use a little help with my treeview. I have roughly the following code:

```
this.#treeview = vscode.window.createTreeView('some-name', {
    treeDataProvider: this.#dataProvider
});

this.#treeview.reveal(myTreeItem).catch(e => {
    console.error(e);
});
```

Unfortunately, revealing the myTreeItem does not work. I get the error: "Tree element not found: [object Object]".
But if I delay the reveal for some time, it works. The dataProvider I am passing is completely sync and the source object already built. Is createTreeView actually async? How do I know when the tree is ready to receive a reveal()?
Or do I have another error in thinking?

**Comment by isidorn** [comment]
@alexr00 can you help here please?

**Comment by alexr00** [comment, marked as answer]
`createTreeView` isn't async, but since we must communicate between the renderer process (which is what causes the UI for your tree view to show) and the extension host (which is where your extension is running) in order to use your tree data provider, if you call `reveal` right after `createTreeView`, you do not have a guarantee that your tree items are already showing.

**Reply by McSodbrenner** [reply]
But then createTreeView is async after all. Otherwise you have no possibility as an extension developer to reveal an item reliably. Is the only way to actually poll? I change in my extension actually often the DataProvider and not always works the reveal.

**Reply by alexr00** [reply]
You can wait for you data provider to get called. Once it's been called and you've returned from `getChildren` then your element will be visible in the tree and safe to reveal.

**Reply by McSodbrenner** [reply]
Thanks a lot. This works but is a little unintuitive. :)

**Comment by Umudhadheni** [comment]
Opening any app
